src/lib/backend/Backend.ts

```diff
@@ -0,0 +1,111 @@
+import type {
+    Author,
+    AuthorSpec,
+    Criterion,
+    CriterionSpec,
+    Paper,
+    PaperSpec,
+    Project,
+    ProjectSpec,
+    ReadingListEntry,
+    Review,
+    ReviewSpec,
+    StageEntry,
+    User,
+    UserSpec,
+} from "./Model";
+
+export interface IUserController {
+    get(): Promise<User>;
+    block(): Promise<void>;
+    changePassword(newPassword: string): Promise<void>;
+    update(newSpec: UserSpec): Promise<User>;
+
+    getProjects(): Promise<Project[]>;
+    getReadingList(): Promise<ReadingListEntry[]>;
+    addToReadingList(paperId: number): Promise<void>;
+    removeFromReadingList(readingListEntryId: number): Promise<void>;
+}
+
+export interface IReviewController {
+    get(): Promise<Review>;
+    set(spec: ReviewSpec): Promise<void>;
+    remove(): Promise<void>;
+}
+
+export interface IPaperController {
+    get(): Promise<Paper>;
+    update(newSpec: PaperSpec): Promise<Paper>;
+
+    getForwardReferencedPapers(): Promise<Paper[]>;
+    getBackwardReferencedPapers(): Promise<Paper[]>;
+}
+
+export interface IStageEntryController {
+    get(): Promise<StageEntry>;
+    remove(): Promise<void>;
+
+    getReviews(): Promise<Review[]>;
+    review(userId: number): IReviewController;
+}
+
+export interface IStageController {
+    getCsv(): Promise<Blob>;
+
+    getPapers(): Promise<StageEntry[]>;
+    addPaper(paperId: number): Promise<void>;
```

Comment:
I think it would be great to return the created `StageEntry` here.
```suggestion
    addPaper(paperId: number): Promise<StageEntry>;
```